hack/generator/pkg/astmodel/type.go

```diff
@@ -35,7 +35,7 @@ type Type interface {
 	AsZero(types Types, ctx *CodeGenerationContext) dst.Expr
 
 	// Equals returns true if the passed type is the same as this one, false otherwise
-	Equals(t Type) bool
+	Equals(t Type, overrides EqualityOverrides) bool
```

Comment:
It seems as though you want most (maybe all?) consumers to use `TypeEquals()` instead of calling this directly. Should we lower the visibility so that consumers outside this package have no choice?